src/settings.cpp

```diff
@@ -1913,6 +1913,12 @@ CommandCost CmdChangeSetting(TileIndex tile, DoCommandFlag flags, uint32 p1, uin
 		}
 
 		SetWindowClassesDirty(WC_GAME_OPTIONS);
+
+#ifdef __EMSCRIPTEN__
```

Comment:
Why not just do this anyway? It's not like saving to file is particularly expensive...

Reply:
I was just ranting about that on IRC :P It is a change in functionality for all OSes ... I am all in favour, but .. I try to minimize the impact with this PR as much as possible :D I will make a PR to apply this for all OSes, and we can see what we think. I can also hide it behind a command line argument ... dealers choice!